src/Compilers/CSharp/Portable/Syntax/ExtensionDeclarationSyntax.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax
+{
+    public partial class ExtensionDeclarationSyntax
+    {
+        public override SyntaxToken Identifier
+            => throw new System.NotImplementedException();
```

Comment:
>throw new System.NotImplementedException(); [](http://example.com/codeflow?start=15&length=43)

I think returning `default` was the right thing to do. A call through base class should be allowed.

Reply:
It is, however, Ok to throw when one tries to set identifier.